### PR TITLE
[`bnb-nightly`] Address final comments

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Address final comments from @ydshieh on. https://github.com/huggingface/peft/pull/1282#pullrequestreview-1791413593 

I left few questions !

cc @ydshieh

### CODE REVIEW

File: .github/workflows/nightly-bnb.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       TEST_TYPE: "single_gpu_${{ matrix.docker-image-name }}"
     container:
       image: ${{ matrix.docker-image-name }}
-      options: --gpus all --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/ -e NVIDIA_DISABLE_REQUIRE=true
+      options: --gpus all --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
```
Comment (younesbelkada) [inline .github/workflows/nightly-bnb.yml:28]: the env variable `NVIDIA_DISABLE_REQUIRE` is set on the very top so these are not needed

File: .github/workflows/nightly-bnb.yml [hunk 2]
```diff
@@ -39,16 +39,19 @@ jobs:
           mkdir transformers-clone && git clone https://github.com/huggingface/transformers.git transformers-clone # rename to transformers clone to avoid modules conflict
 
       - name: Run examples on single GPU
+        if: always()
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply .github/workflows/nightly-bnb.yml:42]: The `fail-fast:` (under `jobs.<job_id>.strategy`) applies to the jobs in the defined matrix. But here each job in the matrix has 3 testing steps in each job, but those steps won't be controlled by `jobs.<job_id>.strategy`.

That is my understanding

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

[the remark this answers is not in the document]
Comment (ydshieh) [reply .github/workflows/nightly-bnb.yml:42]: well, let me double check maybe!

[the remark this answers is not in the document]
Comment (ydshieh) [reply .github/workflows/nightly-bnb.yml:42]: See this example

https://github.com/huggingface/transformers/actions/runs/7286009365/job/19853943708

all 3 jobs (in the matrix) are run. But the 3rd step in each job are not.

File: .github/workflows/nightly-bnb.yml [hunk 3]
```diff
@@ -39,16 +39,19 @@ jobs:
...
         run: |
           source activate peft
           make tests_examples_single_gpu_bnb
       
       - name: Run core tests on single GPU
+        if: always()
```
Comment (ydshieh) [inline .github/workflows/nightly-bnb.yml:48]: This is the quick and simple way. Just want to point out:

- `if: always()` will make the step uncancellable. A better approach would be `if: !cancelled()`

Comment (ydshieh) [reply .github/workflows/nightly-bnb.yml:48]: But I believe we only  want to run the 3 tests only if the step `Pip install` succeed. The condition gets a bit more complex in this case. OK for me if you are happy to keep the current version.

**Review by ydshieh — APPROVED** [review]
LGTM, just give you some info for you to make the final decision.

File: .github/workflows/nightly-bnb.yml (hunk 3 above)
Comment (younesbelkada) [reply .github/workflows/nightly-bnb.yml:48]: Perfect thanks! I will stick to the current plan then :D

**Comment by younesbelkada** [comment]
Thanks a lot for your review @ydshieh !

**Pull request merged**